bin/testEnvLocalK8S.sh

```diff
@@ -128,49 +133,82 @@ function startLocalApiserver() {
     # make sure apiserver is actually alive
     kill -0 $(cat $LOG_DIR/apiserver.pid)
 
-    echo "Started local etcd and apiserver !"
+    # Really need to make sure that API Server is up before proceed further
+    waitForApiServer
+    printf "Started local etcd and apiserver!\n"
+}
+
+function ensureLocalApiServer() {
+    kubectl get nodes 2>/dev/null || startLocalApiserver
+}
+
+function createIstioConfigmap() {
+  helm template ${ISTIO_GO}/install/kubernetes/helm/istio --namespace=istio-system \
+     --execute=templates/configmap.yaml --values install/kubernetes/helm/istio/values.yaml  > /tmp/istio-configmap.yaml
+  kubectl create -f /tmp/istio-configmap.yaml 
+  helm template ${ISTIO_GO}/install/kubernetes/helm/istio --namespace=istio-system \
+     --execute=charts/ingress/templates/service.yaml --values install/kubernetes/helm/istio/values.yaml  > /tmp/istio-ingress.yaml
+  kubectl create -f /tmp/istio-ingress.yaml
 }
 
 function startIstio() {
     ensureLocalApiServer
+    createIstioConfigmap
     startPilot
     startEnvoy
     startMixer
 }
 
 function stopIstio() {
   if [[ -f $LOG_DIR/pilot.pid ]] ; then
-    kill -9 $(cat $LOG_DIR/pilot.pid)
-    kill -9 $(cat $LOG_DIR/mixer.pid)
-    kill -9 $(cat $LOG_DIR/envoy4.pid)
-    rm $LOG_DIR/{pilot,mixer,envoy4}.pid
+    kill -9 $(cat $LOG_DIR/pilot.pid) || true
+    rm $LOG_DIR/pilot.pid
+   fi
+  if [[ -f $LOG_DIR/mixer.pid ]] ; then
+    kill -9 $(cat $LOG_DIR/mixer.pid) || true
+    rm $LOG_DIR/mixer.pid
+  fi
+  if [[ -f $LOG_DIR/envoy4.pid ]] ; then
+    kill -9 $(cat $LOG_DIR/envoy4.pid) || true
+    rm $LOG_DIR/envoy4.pid
   fi
 }
 
 function startPilot() {
-  POD_NAME=pilot POD_NAMESPACE=istio-system ${ISTIO_OUT}/pilot-discovery discovery \
-    -n default --kubeconfig .circleci/config &
+  printf "Pilot starting...\n"
+  POD_NAME=pilot POD_NAMESPACE=istio-system \
+  ${ISTIO_OUT}/pilot-discovery discovery --port 18080 \
+                                         --monitoringPort 19093 \
+                                         --kubeconfig ${ISTIO_GO}/.circleci/config &
   echo $! > $LOG_DIR/pilot.pid
 }
 
 function startMixer() {
-  ${ISTIO_OUT}/mixs server --configStoreURL=fs:${ISTIO_GO}/mixer/testdata/configroot \
-    --kubeconfig .circleci/config &
+  printf "Mixer starting...\n"
+  ${ISTIO_OUT}/mixs server --configStoreURL=fs:${ISTIO_GO}/mixer/testdata/configroot & 
   echo $! > $LOG_DIR/mixer.pid
 }
 
 function startEnvoy() {
+  if [[ "$OSTYPE" != "darwin"* ]]; then
+    printf "Envoy starting...\n"
     ${ISTIO_OUT}/envoy -c tests/testdata/envoy_local.json \
         --base-id 4 --service-cluster unittest --service-node local.test
-  echo $! > $LOG_DIR/envoy4.pid
+    echo $! > $LOG_DIR/envoy4.pid
+  else
+     printf "No Envoy on MAC, sorry...\n"
```

Comment:
Apparently there is one. Piotr got it to compile - he needs to upload it somewhere until we get auto-builds working.

Reply:
great, looking forward for its location.

Reply:
@PiotrSikora could you upload envoy for mac container somewhere so this PR work can continue?  I don't use mac, but alot of folks want to do so, and it would be appreciated.

Cheers
-steve

Reply:
@PiotrSikora Agree about official thing, but if you could get me unofficial just so I could continue playing with it, that would be very nice.

Reply:
@sbezverk yeah, I'm trying to locate the best place to push it... But I'm confused as to how do you want to "play with it" if you're not using mac?

Reply:
@PiotrSikora I don't use Mac for dev, I think @sbezverk does - and perhaps that was confusing about this request.

Reply:
Ooops, indeed, I got confused by the fairly similar usernames, sorry about that!

Anyway, I pushed Istio Proxy built on macOS from 0.7.1 release to https://storage.googleapis.com/istio-on-macos/releases/0.7.1/istio-proxy-0.7.1-macos.tar.gz.

If you're on macOS, then you can build it with:

    $ git checkout 0.7.1
    $ bazel build --config=release //src/envoy:envoy
    ...
    $ openssl sha256 bazel-bin/src/envoy/envoy
    SHA256(bazel-bin/src/envoy/envoy)= 8dbe0adab7761d61ba3a531531e86922a2106567fe1196c638a65fec16a66627

and verify that it matches binary in the linked archive:

    $ curl -Os https://storage.googleapis.com/istio-on-macos/releases/0.7.1/istio-proxy-0.7.1-macos.tar.gz
    $ tar -xOf istio-proxy-0.7.1-macos.tar.gz | openssl sha256
    $ (stdin)= 8dbe0adab7761d61ba3a531531e86922a2106567fe1196c638a65fec16a66627

Reply:
@PiotrSikora @sdake 

We don't build mac binaries in release, so I don't suppose we will have any official macos binaries.

Reply:
@hklai Fair.  I believe @sbezverk was asking for unofficial builds so he could make forward progress on this PR.  Not sure though how other people would consume this PR without official Mac builds of proxy.

Cheers
-steve

Reply:
@PiotrSikora @costinm Thanks a lot for the link, when I try to start envoy using envoy_local.json, I see the following errors:

```
Envoy starting...
[2018-04-10 11:01:55.957][12685001][info][main] external/envoy/source/server/server.cc:178] initializing epoch 0 (hot restart version=disabled)
[2018-04-10 11:01:55.962][12685001][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:127] cm init: initializing cds
[2018-04-10 11:01:55.963][12685001][info][config] external/envoy/source/server/configuration_impl.cc:52] loading 6 listener(s)
[2018-04-10 11:01:55.965][12685001][critical][main] external/envoy/source/server/server.cc:71] error initializing configuration 'tests/testdata/envoy_local.json': route: unknown cluster 'service1'
[2018-04-10 11:01:55.965][12685001][info][main] external/envoy/source/server/server.cc:392] exiting
```
I have not modified this file I was under assumption that since it is in repo it should be good but apparently it is not, appreciate if you could review it and let me know what cab be done.